ArduCopter/mode_auto.cpp

```diff
@@ -343,63 +356,59 @@ void ModeAuto::land_start(const Vector2f& destination)
     copter.ap.prec_land_active = false;
 }
 
-// auto_circle_movetoedge_start - initialise waypoint controller to move to edge of a circle with it's center at the specified location
-//  we assume the caller has performed all required GPS_ok checks
-void ModeAuto::circle_movetoedge_start(const Location &circle_center, float radius_m)
+Mode::_LoiterTurns Mode::loiter_turns;
+
+void Mode::_LoiterTurns::initialise(const Location &circle_center_loc, float circle_radius_m, uint8_t turns)
 {
     // set circle center
-    copter.circle_nav->set_center(circle_center);
+    copter.circle_nav->set_center(circle_center_loc);
 
     // set circle radius
-    if (!is_zero(radius_m)) {
-        copter.circle_nav->set_radius(radius_m * 100.0f);
-    }
```

Comment:
We appear to have lost this clause.

Reply:
This is intentional. As mentioned issue #19028, we should not ignore 0 radius rather we should rotate at center for 0 radius.  Hence, we needed to remove this `if(!is_zero(radius_m))` check and initialise radius even for 0 value.

Reply:
You should *most definitely* avoid changing behaviour like this in a PR which unrelated to fixing that bug!

A separate PR to fix that would be fine - and then you rebase this PR on that PR.

Why?  Because I instantly flagged the loss of that clause in this PR considering it breakage!  That's now cost both of us time.... thus keeping things separated.

Reply:
Sure! Actually @muramura asked me to test operation with zero radius on this PR. It didn't work fine and I discovered #19028. So thought of fixing it to make him happy :-). But yeah, I'll take care of this in future. Thanks for letting me know.